lib/utility/_query.js

```diff
@@ -50,19 +50,43 @@ exports._query = function _query(query, data, options, callback, executor) {
 
 	logger.trace('ORACLE QUERY=>', query, data);
 
-	this.getConnection(function (err, connection) {
-		if (err) {
-			return callback(err);
-		}
-
+	/**
+	 * Execute query over db.
+	 */
+	var executeQuery = function (connection, reconnect) {
 		/**
 		 * Row prefetching allows multiple rows to be returned from Oracle Database to Node.js in each
 		 * network round-trip. We can control how many rows are prefetched via the prefetchRows property
 		 * of either the base driver class or the options object passed to the execute method of a connection.
 		 */
```

Comment:
Please move that comment up to the if block starting in Line 42. It's not clear what that comment refers to now that the prefetch function if further down.